WordPressUI/Bottom Sheet/DrawerPresentationController.swift

```diff
@@ -0,0 +1,522 @@
+
+    /// Returns the Y position of the drawer
+    public var yPosition: CGFloat? {
+        return presentedView?.frame.origin.y
+    }
+
+    /// Animates between the drawer positions
+    /// - Parameter position: The position to animate to
+    public func transition(to position: DrawerPosition) {
+        currentPosition = position
+
+        if position == .closed {
+            dismiss()
+            return
+        }
+
+        var margin: CGFloat = 0
+
+        switch position {
+        case .expanded:
+            margin = expandedYPosition
+
+        case .collapsed:
+            margin = collapsedYPosition
+
+        case .hidden:
+            margin = hiddenYPosition
+
+        default:
+            margin = 0
+        }
+
+        setTopMargin(margin)
+    }
+
+    @objc func dismiss() {
+        presentedViewController.dismiss(animated: true, completion: nil)
+    }
+
+    public override func presentationTransitionWillBegin() {
+        super.presentationTransitionWillBegin()
+
+        configureScrollViewInsets()
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        transition(to: currentPosition)
+    }
+
+    public override func presentationTransitionDidEnd(_ completed: Bool) {
+        super.presentationTransitionDidEnd(completed)
+
+        configureScrollViewInsets()
+    }
+
+
+    // MARK: - Internal Positions
+    // Helpers to calculate the Y positions for the drawer positions
+
+    private var closedPosition: CGFloat {
+        guard let presentedView = self.presentedView else {
+            return 0
+        }
+
+        return presentedView.bounds.height
+    }
+
+    private var collapsedYPosition: CGFloat {
+        let height = presentableViewController?.collapsedHeight ?? Constants.Defaults.collapsedHeight
+
+        return topMargin(with: height)
+    }
+
+    private var expandedYPosition: CGFloat {
+        let height = presentableViewController?.expandedHeight ?? Constants.Defaults.expandedHeight
+
+        return topMargin(with: height)
+    }
+
+    private var hiddenYPosition: CGFloat {
+        return topMargin(with: .hidden)
+    }
+
+    /// Calculates the Y position for the view based on a DrawerHeight enum
+    /// - Parameter drawerHeight: The drawer height to calculate
+    private func topMargin(with drawerHeight: DrawerHeight) -> CGFloat {
+        var topMargin: CGFloat
+
+        switch drawerHeight {
+        case .contentHeight(let height):
+            topMargin = calculatedTopMargin(for: height)
+
+        case .topMargin(let margin):
+            topMargin = safeAreaInsets.top + margin
+
+        case .maxHeight:
+            topMargin = safeAreaInsets.top
+
+        case .intrinsicHeight:
+            let height = presentedViewController.children.first?.preferredContentSize.height ?? 0 + BottomSheetViewController.Constants.additionalContentTopMargin
```

Comment:
This works well, but I don't think we should reference the `BottomSheetViewController` in this class.

Reply:
Very good point.

Changed in 69352ab